### PR TITLE
Add identifiers to import schema

### PR DESCRIPTION
**Pull request by jimchamp**
## Description:
Adds optional `identifiers` field to our import object schema.  `identifiers` is an object that maps lists of stings to a string key.

### CODE REVIEW

**Comment by hornc** [comment]
@jimchamp It looks like this is meant to be the same as https://github.com/internetarchive/openlibrary-client/blob/master/olclient/schemata/edition.schema.json#L145-L147 which could probably do with a `patternProperties` too.

There isn't much evidence that the code does anything with `identifiers` in https://github.com/internetarchive/openlibrary/blob/master/openlibrary/plugins/importapi/import_edition_builder.py , and `identifiers` is not even listed as an edition property at https://openlibrary.org/type/edition , but I _believe_ the importer will pass the `identifiers` on to the created edition -- have you confirmed this?
**edit:** I see you have added tests with `identifiers` in https://github.com/internetarchive/openlibrary/blob/master/openlibrary/catalog/add_book/tests/test_add_book.py , so I think that's covered 👍🏽  


`identifiers` do belong on editions, so if the import works, then this is correct. We could be explicit about which identifiers we do use -- some examples might be helpful. It's somewhat confusing that `ocaid` or `isbn` would be incorrect under this property.

`goodreads`, `librarything` are the common ones, I think I've used `eebo` , but now I can't find where the drop-down values are located.

**Comment by jimchamp** [comment]
@hornc Thanks for the helpful notes!

I have confirmed that the importer is passing `identifiers` onto the created editions. I've also added some examples and a brief description of this property in the import schema definition.

> `goodreads`, `librarything` are the common ones, I think I've used `eebo` , but now I can't find where the drop-down values are located.

I've been using this as the source of truth for edition `identifiers`: https://openlibrary.org/config/edition.yml?m=edit

**Comment by jimchamp** [comment]
I see now that I should have provided some more details when I opened the PR.  Here is a bit of background information:

This change was made in support of earlier and more thorough validations in our import flow.  Ultimately, we would like to use the import schema to generate Pydantic classes which will be used to list all of a record's errors in a single validation call (as opposed to `jsonschema.validate()`, which can fail with new errors after initial validation errors have been corrected).

Hopefully this will allow us to reduce the amount of low quality records being created, and shorten the total run time of our import jobs.

File: olclient/schemata/import.schema.json [hunk 1]
```diff
@@ -93,6 +93,18 @@
         "12 ounces",
         "1 pounds"
       ]
+    },
+    "identifiers": {
+      "type": "object",
+      "patternProperties": {
+        "^\\w+": { "$ref": "shared_definitions.json#/string_array" }
+      },
+      "description": "Unique identifiers used by external sites to identify a book. Useful for linking to other sites.",
```
Comment (jimchamp) [inline olclient/schemata/import.schema.json:102]:
```suggestion
      "description": "Unique identifiers used by external sites to identify a book. Used by Open Library to link offsite.",
```

File: olclient/schemata/import.schema.json [hunk 2]
```diff
@@ -93,6 +93,18 @@
...
+      "type": "object",
+      "patternProperties": {
+        "^\\w+": { "$ref": "shared_definitions.json#/string_array" }
+      },
+      "description": "Unique identifiers used by external sites to identify a book. Used by Open Library to link offsite.",
+      "examples": [
+        "standard_ebooks",
+        "project_gutenberg",
+        "wikisource"
+      ]
```
Comment (cdrini) [inline olclient/schemata/import.schema.json:107]: Expand the examples to be full

```suggestion
      "examples": [
        {
            "standard_ebooks": ["leo-tolstoy/what-is-art/aylmer-maude"]
        },
        {
            "project_gutenberg": ["64317"]
        }
      ]
```

**Review by cdrini — APPROVED** [review]
Seems like @hornc 's concerns have been addressed! I tested:

```python
import jsonschema
import os
import json

schemata_path = os.path.join(os.getcwd(), 'olclient', 'schemata', 'import.schema.json')
schema = json.load(open(schemata_path))
resolver = jsonschema.RefResolver('file://' + schemata_path, schema)
jsonschema.Draft4Validator(schema, resolver=resolver).validate({
    'title': 'B',
    'source_records': ['blah:23425'],
    'authors': [{'name': 'Jim'}],
    'publishers': ['OL Co.'],
    'publish_date': '1234',
    'identifiers': {'goodreads': ['1234']},
})
```

And worked like a charm!

**Comment by cdrini** [comment]
Fun fact: TIL I learned you can use gitpod even on openlibrary-client! Even though we haven't configured it! Just type `gitpod.io/#` before the URL of a PR or the main repo to jump into a gitpod environment :) (Note to bring up a terminal, press `ctrl+p`, and type `> terminal`)

**Comment by cclauss** [comment]
Please demo the GitPod thing in today’s community meeting.

**Pull request merged**